# PeptipediaDB: peptide sequence database and user-friendly web platform.


This repository contains the source files and supplementary information for the implementations and use cases presented in the work: <br>
Gabriel Cabas-Mora<sup>1</sup>, Anamaría Daza<sup>2</sup>, Lindybeth Sarmiento-Varón<sup>3</sup>, Diego Alvarez<sup>3,4</sup>, Valentina Garrido<sup>1</sup>, Julieta H. Sepúlveda<sup>5</sup>, Roberto Uribe-Paredes<sup>1</sup>, Álvaro Olivera-Nappa<sup>2</sup>, Mehdi D. Davari<sup>6</sup>, Marcelo Navarrete<sup>3,4</sup> and David Medina-Ortiz<sup>1,2*</sup>.<br>

PeptipediaDB: peptide sequence database and user-friendly web platform. A major update. <br>

https://doi.org/XXXX<br>

<sup>*1*</sup><sub>Departamento de Ingeniería en Computación, Universidad de Magallanes, Av. Pdte. Manuel Bulnes 01855, 6210427, Punta Arenas, Chile.</sub> <br>
<sup>*2*</sup><sub>Centre for Biotechnology and Bioengineering, CeBiB, Universidad de Chile, Avenida Beauchef 851, 8320000, Santiago, Chile.</sub> <br>
<sup>*3*</sup><sub>Centro Asistencial de Docencia e Investigación, CADI, Universidad de Magallanes, Av. Los Flamencos 01364, 6210005, Punta Arenas, Chile.</sub> <br>
<sup>*4*</sup><sub>Escuela de Medicina, Universidad de Magallanes, Av. Pdte. Manuel Bulnes 01855, 6210427, Punta Arenas, Chile.</sub> <br>
<sup>*5*</sup><sub>Facultad de Ciencias de la Salud, Universidad de Magallanes, Av. Pdte. Manuel Bulnes 01855, 6210427, Punta Arenas, Chile.</sub> <br>
<sup>*6*</sup><sub>Department of Bioorganic Chemistry, Leibniz Institute of Plant Biochemistry, Weinberg 3, 06120, Halle, Germany.</sub> <br>
<sup>*\**</sup><sub>Corresponding author</sub> <br>

---
## Table of Contents
- [A summary of the proposed work](#summary)
- [Requirements and instalation](#requirements)
---

<a name="summary"></a>

# PeptipediaDB: peptide sequence database and user-friendly web platform. A major update.

Peptides have gained greater relevance in recent years thanks to their therapeutic properties. The increase in the production and synthesis of peptides has resulted in a large volume of data, allowing the generation of databases and information repositories. Significant advances in sequencing techniques and artificial intelligence aimed at accelerating peptide design. However, applying these techniques requires versatile and constantly updated storage systems, along with tools that facilitate peptide research and the application of machine learning techniques for building predictive systems. In this work, we present a significant update of our Peptipedia database, increasing by more than 45% the sequences with experimentally validated biological activity and more than 3.9 million peptides with biological activity predicted through machine learning models. All peptide sequences are described using physicochemical, thermodynamic, structural, and ontologic descriptions. Finally, peptide description tools are incorporated through structural and ontological properties, predictive models of relevant properties to peptide design, and more than 70 binary biological classification models are added along with a moonlight effect estimation system. This new Peptipedia version represents the most significant public repository of peptides and facilitates the study of peptides as support for biotechnological research. Peptipedia is publicly accessible on https://peptipedia.cl/ for non-commercial use licensed under the MIT License.

This repository contains the Peptitools application, a web application for studying peptide sequences.

<a name="requirements"></a>

## Requirements and instalation.

This web application was implemented using a client-server architecture. The frontend and backend folder contains information about requirements and instalation in his own README.md.

## Tools.

Peptitools allows you to run:

- Bioinformatic analysis in your own peptide sequences like: Multiple Sequences Alignments, pfam domains, gene ontology terms and secondary structure predictions.

- Statistics tests in grouped sequences, comparing physicochemical properties and residues distributions.

- Pipelines of supervised and non-supervised applications, using numerical representation techniques and various enabled algorithms.

- Evaluation and prediction using our published predictive models.